modules/generators/content/bash_history/secgen_local/local.rb

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/ruby
+require_relative '../../../../../lib/objects/local_string_generator.rb'
+require 'erb'
+require 'fileutils'
+class BashHistoryGenerator < StringGenerator
+  attr_accessor :command_sample
+  attr_accessor :sudo_sample
+  attr_accessor :password_sample
+  LOCAL_DIR = File.expand_path('../../',__FILE__)
+  TEMPLATE_PATH = "#{LOCAL_DIR}/templates/bash_history.md.erb"
+
+  def generate
+
+    sudo_array = File.readlines('../../../../../lib/resources/linelists/top_50_sudo_commands')
+    self.sudo_sample = sudo_array.sample(5)
+    password_array = File.readlines('../../../../../lib/resources/wordlists/10_million_password_list_top_100')
```

Comment:
Read the password via a parameter (aka input, aka fact)

To do that you need to define the parameter(s) in the secgen_metadata.xml:
```
<read_fact>password</read_fact>
```
Then read the input in the puppet code (local.rb):
```
  def get_options_array
    super + [['--password', GetoptLong::OPTIONAL_ARGUMENT]]
  end

  def process_options(opt, arg)
    super
    case opt
      when '--password'
        self.password_sample << arg;
    end
  end
```